src/Analyzer/Resolve/QueryAnalyzer.cpp

```diff
@@ -1650,1079 +1175,67 @@ QueryTreeNodePtr QueryAnalyzer::tryResolveIdentifierFromAliases(const Identifier
     return alias_node;
 }
 
-/** Resolve identifier from table columns.
+/** Try resolve identifier in current scope parent scopes.
   *
-  * 1. If table column nodes are empty or identifier is not expression lookup return nullptr.
-  * 2. If identifier full name match table column use column. Save information that we resolve identifier using full name.
-  * 3. Else if identifier binds to table column, use column.
-  * 4. Try to resolve column ALIAS expression if it exists.
-  * 5. If identifier was compound and was not resolved using full name during step 1 use `tryResolveIdentifierFromCompoundExpression`.
-  * This can be the case with compound ALIAS columns.
+  * TODO: If column is matched, throw exception that nested subqueries are not supported.
   *
-  * Example:
-  * CREATE TABLE test_table (id UInt64, value Tuple(id UInt64, value String), alias_value ALIAS value.id) ENGINE=TinyLog;
+  * If initial scope is expression. Then try to resolve identifier in parent scopes until query scope is hit.
+  * For query scope resolve strategy is same as if initial scope if query.
   */
-QueryTreeNodePtr QueryAnalyzer::tryResolveIdentifierFromTableColumns(const IdentifierLookup & identifier_lookup, IdentifierResolveScope & scope)
+IdentifierResolveResult QueryAnalyzer::tryResolveIdentifierInParentScopes(const IdentifierLookup & identifier_lookup, IdentifierResolveScope & scope)
```

Comment:
Why does it stay?

Reply:
Could not move it now, because of the dependency chain:
`tryResolveIdentifierInParentScopes -> tryResolveIdentifier -> tryResolveIdentifierFromAliases -> resolveExpressionNode`